### PR TITLE
Update perf test suite

### PR DESCRIPTION
**Pull request by yf225**
This PR updates various aspects of the perf test suite:

- Hard exit when test script outputs warning or error, so that we can go in and fix them earlier
- Reduced the allowed range of z-value, since the runtime of CPU perf tests are more stable now (by running them on OS X workers)
- Moved perf test scripts to PyTorch account

### CODE REVIEW

File: .jenkins/perf_test/test_cpu_speed_mini_sequence_labeler.sh [hunk 1]
```diff
@@ -6,17 +6,25 @@ test_cpu_speed_mini_sequence_labeler () {
   export OMP_NUM_THREADS=4
   export MKL_NUM_THREADS=4
 
-  curl https://gist.githubusercontent.com/yf225/40c0adb8bfb2a7b774fa266fb4bc409e/raw/20c67ebadbd75f41c6c9fd2e00b4b2562b60700a/mini_sequence_labeler.py -O
-  curl https://gist.githubusercontent.com/yf225/592b39ca6a3fc835a4d1532fb1474d26/raw/76f57c198cb7afdc5122e413c2a3023ed024b643/wsj.pkl -O
+  git clone https://github.com/pytorch/benchmark.git
+
+  cd benchmark/
+
+  git fetch origin pull/14/head:mini_sequence_labeler
```
Comment (ezyang) [inline .jenkins/perf_test/test_cpu_speed_mini_sequence_labeler.sh]: Shouldn't we merge it first? :)

File: .jenkins/perf_test/perf_test_numbers.json [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
-		"sigma": "2.37561"
+		"mean": "39.31513",
+		"sigma": "1.44542"
 	},
 
+	"gpu-test-base-machine": "g3.8xlarge",
+
 	"test_gpu_speed_mnist": {
-		"mean": "13.76155",
-		"sigma": "0.42087"
```
Comment (ezyang) [inline .jenkins/perf_test/perf_test_numbers.json]: We're intentionally omitting these numbers here?

**Comment by ezyang** [comment]
One comment; assuming that it's intentional this is good to merge.

File: .jenkins/perf_test/perf_test_numbers.json (hunk 2 above)
Comment (yf225) [reply .jenkins/perf_test/perf_test_numbers.json]: Yes it's intentional - I haven't collected the baseline numbers for GPU yet and will open a PR for it later.

**Pull request merged**